pkg/client/dtclient/settings_client.go

```diff
@@ -563,24 +563,48 @@ func doObjectsMatchBasedOnUniqueKeys(uniqueKeys []string, source SettingsObject,
 	return true, matches, nil
 }
 
-func isSameValueForKey(key string, c1 []byte, c2 []byte) (same bool, matchingVal any, err error) {
+func isSameValueForKey(targetPath string, c1 []byte, c2 []byte) (same bool, matchingVal any, err error) {
 	u := make(map[string]any)
 	if err := json.Unmarshal(c1, &u); err != nil {
-		return false, nil, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+		return false, nil, fmt.Errorf("failed to unmarshal data for key %q: %w", targetPath, err)
 	}
-	v1 := u[key]
 
+	keys := explodePath(targetPath)
+	value1 := recursiveSearch(u, keys)
+
+	u = nil
```

Comment:
Instead of reusing `u`, I'd go with a new var. Setting it to `null` is pointless, as an error is returned anyway if it can't be unmarshaled.